photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoEditorImpl.java

```diff
@@ -343,4 +355,12 @@ public void setOnPhotoEditorListener(@NonNull OnPhotoEditorListener onPhotoEdito
     public boolean isCacheEmpty() {
         return viewState.getAddedViewsCount() == 0 && viewState.getRedoViewsCount() == 0;
     }
-}
\ No newline at end of file
+
+    // region Shape
+    @Override
+    public void updateShape(ShapeBuilder shapeBuilder) {
```

Comment:
This will add the shape. But the undo/redo won't work.

Reply:
Undo/redo works again in the new commit.